types/stylelint/index.d.ts

```diff
@@ -53,574 +56,741 @@ type RuleMessageFunc = {
 	};
+	reportedDisables: DisableOptionsReport;
+	descriptionlessDisables?: DisableOptionsReport;
+	needlessDisables?: DisableOptionsReport;
+	invalidScopeDisables?: DisableOptionsReport;
+	/**
+	 * Each rule metadata by name.
+	 */
+	ruleMetadata: { [ruleName: string]: Partial<RuleMeta> };
+};
 
+/**
+ * A lint problem.
+ */
+export type Problem = {
+	ruleName: string;
+	result: PostcssResult;
+	message: RuleMessage;
+	messageArgs?: Parameters<RuleMessageFunc> | undefined;
+	node: PostCSS.Node;
 	/**
-	 * A CSS syntax error.
+	 * The inclusive start index of the problem, relative to the node's
+	 * source text.
 	 */
-	type CssSyntaxError = {
-		file?: string;
-		input: {
-			column: number;
-			file?: string;
-			line: number;
-			source: string;
-		};
-		/**
-		 * The line of the inclusive start position of the error.
-		 */
+	index?: number;
+	/**
+	 * The exclusive end index of the problem, relative to the node's
+	 * source text.
+	 */
+	endIndex?: number;
+	/**
+	 * The inclusive start position of the problem, relative to the
+	 * node's source text. If provided, this will be used instead of
+	 * `index`.
+	 */
+	start?: {
 		line: number;
-		/**
-		 * The column of the inclusive start position of the error.
-		 */
 		column: number;
-		/**
-		 * The line of the exclusive end position of the error.
-		 */
-		endLine?: number;
-		/**
-		 * The column of the exclusive end position of the error.
-		 */
-		endColumn?: number;
-		message: string;
-		name: string;
-		reason: string;
-		source: string;
 	};
-
 	/**
-	 * A lint warning.
+	 * The exclusive end position of the problem, relative to the
+	 * node's source text. If provided, this will be used instead of
+	 * `endIndex`.
 	 */
-	type Warning = {
-		/**
-		 * The line of the inclusive start position of the warning.
-		 */
+	end?: {
 		line: number;
-		/**
-		 * The column of the inclusive start position of the warning.
-		 */
 		column: number;
-		/**
-		 * The line of the exclusive end position of the warning.
-		 */
-		endLine?: number;
-		/**
-		 * The column of the exclusive end position of the warning.
-		 */
-		endColumn?: number;
-		rule: string;
-		severity: Severity;
-		text: string;
-		stylelintType?: string;
 	};
-
+	word?: string;
+	line?: number;
 	/**
-	 * A lint result.
-	 */
-	type LintResult = {
-		source?: string;
-		deprecations: {
-			text: string;
-			reference?: string;
-		}[];
-		invalidOptionWarnings: {
-			text: string;
-		}[];
-		parseErrors: (PostCSS.Warning & { stylelintType: string })[];
-		errored?: boolean;
-		warnings: Warning[];
-		ignored?: boolean;
-		/**
-		 * Internal use only. Do not use or rely on this property. It may
-		 * change at any time.
-		 * @internal
-		 */
-		_postcssResult?: PostcssResult;
-	};
-
-	/** @internal */
-	type DisableReportRange = {
-		rule: string;
-		start: number;
-		end?: number;
-	};
+	 * Optional severity override for the problem.
+	 */
+	severity?: RuleSeverity;
+};
 
+/** @internal */
+export type ShorthandProperties =
```

Comment:
minor
non-blocking

This union should have been generated from a `Set` imported from a file in the reference directory.

Reply:
Yeah, syncing the implementation and type definition is best. I don't know how to achieve it, though. 😅

Reply:
It should look something like that:

```
export const bar = /** @type {const} */ (['animation', 'background', 'etc']);
…
import { bar } from 'qux.mjs';
type Baz = typeof foo[number]; // 'animation' | 'background' | 'etc'
```

Reply:
Let's discuss this in another place.